### PR TITLE
Use fetch API in InfluxDB data source

### PR DESCRIPTION
**Pull request by dprokop**
Replaces usages of datasourceRequest in InfluxDB data source in favor of fetch API.

Ref https://github.com/grafana/grafana/issues/27222

### CODE REVIEW

File: public/app/plugins/datasource/influxdb/datasource.ts [hunk 1]
```diff
@@ -151,54 +153,56 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
...
-      const seriesList = [];
-      for (i = 0; i < data.results.length; i++) {
-        const result = data.results[i];
-        if (!result || !result.series) {
-          continue;
+    return this._seriesQuery(allQueries, options)
+      .toPromise()
+      .then((data: any): any => {
+        if (!data || !data.results) {
+          return [];
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/influxdb/datasource.ts:160]: I think classicQuery should return an Observable instead and then we could remove __from__ creator here https://github.com/grafana/grafana/blob/316a7997a53882fe6acb37f4cffa0757bb674012/public/app/plugins/datasource/influxdb/datasource.ts#L80

and then we don't need to do `toPromise` here. WDTY?

File: public/app/plugins/datasource/influxdb/datasource.ts [hunk 2]
```diff
@@ -459,46 +468,59 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
...
-                throw {
-                  message: 'InfluxDB Error: ' + errors[0].error,
-                  data,
-                };
+                return throwError(
+                  this.handleErrors({
+                    message: 'InfluxDB Error: ' + errors[0].error,
+                    data,
+                  })
+                );
```
Comment (hugohaggmark) [inline public/app/plugins/datasource/influxdb/datasource.ts:486]: This is returning an Observable but the other path is returning an array. Maybe use a simple `throw` instead?

**Review by hugohaggmark — COMMENTED** [review]
Skilled work on this! I left some considerations I think could improve this.

**Comment by dprokop** [comment]
@hugohaggmark thank you for your review. I've applied the changes you suggested!

**Review by hugohaggmark — APPROVED** [review]
LGTM!

**Pull request merged**